rcl_logging_spdlog/src/rcl_logging_spdlog.cpp

```diff
@@ -53,6 +55,42 @@ static spdlog::level::level_enum map_external_log_level_to_library_level(int ext
   return level;
 }
 
+namespace
+{
+
+RCL_LOGGING_INTERFACE_LOCAL
+bool
+get_should_use_old_flushing_behavior()
+{
+  const char * env_var_name = "RCL_LOGGING_SPDLOG_EXPERIMENTAL_OLD_FLUSHING_BEHAVIOR";
+
+  try {
+    std::string env_var_value = rcpputils::get_env_var(env_var_name);
+
+    if (env_var_value.empty()) {
+      // not set
+      return false;
+    }
+    if ("0" == env_var_value) {
+      // explicitly false
+      return false;
+    }
+    if ("1" == env_var_value) {
+      // explicitly true
+      return true;
+    }
+
+    // unknown value
+    throw std::runtime_error("unrecognized value: " + env_var_value);
```

Comment:
maybe printing a warning/error and `return false;`
It seems too much to hard fail because the value of an environment variable is unexpected, but I don't mind strongly

Reply:
It was just easier to throw and catch this (it's a private function). It sets the error state where it is caught.

Reply:
The difference is that now `rcl_logging_external_initialize()` is failing, returning `RCL_LOGGING_RET_ERROR`.
And I'm suggesting to log an error and ignore (`should_use_old_flushing_behavior = false` in that case).
Though it may be confusing if someone sets the env variable to "true" and doesn't see the log, so maybe the current behavior is better.